src/python/Utils/Throttled.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import cherrypy
+import threading
+
+class _ThrottleCounter(object):
+
+    def __init__(self, throttle, user):
+        self.throttle = throttle
+        self.user = user
+
+    def __enter__(self):
+        ctr = self.throttle._incUser(self.user)
+        #self.throttle.logger.debug("Entering throttled function with counter %d for user %s" % (ctr, self.user))
+        if ctr >= self.throttle.getLimit():
+            self.throttle._decUser(self.user)
+            raise ExecutionError("The current number of active operations for this resource exceeds the limit of %d for user %s" % (self.throttle.getLimit(), self.user))
+
+    def __exit__(self, type, value, traceback):
```

Comment:
Do we actually need those 3 args? If we do, it would be great to rename `type` to something else.

Reply:
The `__exit__` python API is used when someone wants to have class works with `with` statement. Therefore it requires 3 arguments, the exception type, value and traceback. You can read more about `__enter__` and `__exit__` implementation here: https://stackoverflow.com/questions/1984325/explaining-pythons-enter-and-exit
But I agree that usage of reserved keyword `type` should be avoid. Instead the argument keywords should be changed to `self, exc_type, exc_value, exc_traceback`. I'll this change.